src/markdown/generator.ml

```diff
@@ -0,0 +1,310 @@
+             match content with
+             | [] -> noop
+             | i :: rest -> (
+                 match i.desc with
+                 | Text s ->
+                     command "" (str "[%s]" s ++ str "(%s)" href) ++ f rest
+                 | _ -> noop ++ f rest)
+           in
+           f content)
+          ++ inline rest
+      | InternalLink (Resolved (link, content)) -> (
+          match link.page.parent with
+          | Some _ -> inline content ++ inline rest
+          | None ->
+              open_sq_bracket ++ inline content ++ close_sq_bracket
+              ++ open_parenthesis
+              ++ String ("#" ^ link.anchor)
+              ++ close_parenthesis ++ inline rest)
+      | InternalLink (Unresolved content) -> inline content ++ inline rest
+      | Source content -> env Any "`" "`" (source_code content) ++ inline rest
+      | Raw_markup t -> raw_markup t ++ inline rest)
+
+let rec block (l : Block.t) =
+  match l with
+  | [] -> noop
+  | b :: rest -> (
+      let continue r = if r = [] then noop else break ++ block r in
+      match b.desc with
+      | Inline i -> inline i ++ continue rest
+      | Paragraph i -> inline i ++ continue rest
+      | List (list_typ, l) ->
+          let f n b =
+            let bullet =
+              match list_typ with
+              | Unordered -> escaped "- "
+              | Ordered -> str "%d) " (n + 1)
+            in
+            bullet ++ block b ++ break
+          in
+          list ~sep:break (List.mapi f l) ++ continue rest
+      | Description _ ->
+          let descrs, _, rest =
+            Take.until l ~classify:(function
+              | { Block.desc = Description l; _ } -> Accum l
+              | _ -> Stop_and_keep)
+          in
+          let f i =
+            let key = inline i.Description.key in
+            let def = block i.Description.definition in
+            break ++ str "@" ++ key ++ str " : " ++ def ++ break ++ break
+          in
+          list ~sep:break (List.map f descrs) ++ continue rest
+      | Source content -> source_code content ++ continue rest
+      | Verbatim content -> str "%s" content ++ continue rest
+      | Raw_markup t -> raw_markup t ++ continue rest)
+
+let expansion_not_inlined url = not (Link.should_inline url)
+
+let take_code l =
+  let c, _, rest =
+    Take.until l ~classify:(function
+      | DocumentedSrc.Code c -> Accum c
+      | DocumentedSrc.Alternative (Expansion e) when expansion_not_inlined e.url
+        ->
+          Accum e.summary
+      | _ -> Stop_and_keep)
+  in
+  (c, rest)
+
+let heading { Heading.label; level; title } =
+  let level =
+    match level with
+    | 1 -> "#"
+    | 2 -> "##"
+    | 3 -> "###"
+    | 4 -> "####"
+    | 5 -> "#####"
+    | 6 -> "######"
+    | _ -> ""
+    (* We can be sure that h6 will never be exceded! *)
+  in
+  match label with
+  | Some _ -> command level (str " " ++ inline title)
+  | None -> command (level ^ " ") (inline title)
```

Comment:
Not sure what's the difference between those two, but in any case, the label is missing. There is a pandoc extension for labels.

Reply:
This extension is [header_attributes](https://pandoc.org/MANUAL.html#extension-header_attributes).
We might need to add a "target" option (with values either `pandoc` or `gh` let's say) and to use the syntax only if it makes sense.

Anyway, for now, we need to at least use the `<a id="...">` trick to make the label work.

Reply:
There is no `<a id...>` on headings with `--generate-links` enabled.